dev/ToastNotifications/ToastNotificationUtility.cpp

```diff
@@ -212,3 +212,80 @@ std::wstring RegisterComActivatorGuidAndAssets(winrt::Microsoft::Windows::ToastN
     THROW_IF_WIN32_ERROR(status);
     return registeredGuid;
 }
+
+wil::unique_cotaskmem_string ConvertUtf8StringToWideString(unsigned long length, const byte* utf8String)
+{
+    int size{ MultiByteToWideChar(
+        CP_UTF8,
+        0,
+        reinterpret_cast<PCSTR>(utf8String),
+        length,
+        nullptr,
+        0) };
+    THROW_LAST_ERROR_IF(size == 0);
+
+    wil::unique_cotaskmem_string wideString{ wil::make_unique_string<wil::unique_cotaskmem_string>(nullptr, size) };
+
+    size = MultiByteToWideChar(
+        CP_UTF8,
+        0,
+        reinterpret_cast<PCSTR>(utf8String),
+        length,
+        wideString.get(),
+        size);
+    THROW_LAST_ERROR_IF(size == 0);
+
+    return wideString;
+}
+
+winrt::Microsoft::Windows::ToastNotifications::ToastNotification ToastNotificationFromToastProperties(ABI::Microsoft::Internal::ToastNotifications::INotificationProperties* properties)
+{
+    unsigned int payloadSize{};
+    wil::unique_cotaskmem_array_ptr<byte> payload;
+    properties->get_Payload(&payloadSize, &payload);
```

Comment:
Just don't forget to add THROW macros.

Reply:
Right, added them to calls made into the UDK